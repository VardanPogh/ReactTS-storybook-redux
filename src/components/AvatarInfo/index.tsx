import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import Avatar from '@material-ui/core/Avatar'

const useStyles = makeStyles((theme) => ({
  large: {
    width: theme.spacing(28),
    height: theme.spacing(28),
    margin: 'auto',
  },
  p_name: {
    color: '#185ab4;',
    fontSize: 24,
    marginBottom: 10,
    marginTop: 10,
    textAlign: 'center',
  },
  p_des: {
    fontSize: 18,
    color: '#545454',
    maxWidth: 1000,
    margin: '0 auto',
    textAlign: 'center',
  },
}))

interface Props {
  image: any
  text: string
  title: string
}

const AvatarInfo: React.FC<Props> = ({ image, text, title }) => {
  const classes = useStyles()

  return (
    <>
      <Avatar src={image} className={classes.large} />
      <p className={classes.p_name}>{title}</p>
      <p className={classes.p_des}>{text}</p>
    </>
  )
}

export default AvatarInfo
