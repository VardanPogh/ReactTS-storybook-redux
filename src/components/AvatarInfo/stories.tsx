import React from 'react'
import { withKnobs, text, files } from '@storybook/addon-knobs'

import AvatarInfo from '.'

export default {
  title: 'AvatarInfo',
  component: AvatarInfo,
  decorators: [withKnobs],
}
const info =
  'Ut interdum lectus ut velit posuere auctor. Nam non ultrices orci. Vestibulum nisl neque, condimentum eget justo vel, sagittis lacinia dolor. Vestibulum sapien lectus, ullamcorper in sem eget, cursus accumsan lacus.'

export const Default: React.FC = () => (
  <AvatarInfo
    image={files('image', '', [])}
    title={text('title', 'Dr Jonh Smith')}
    text={text('text', info)}
  />
)
