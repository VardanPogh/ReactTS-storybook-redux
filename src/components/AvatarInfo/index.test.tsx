import React from 'react'
import { render } from '@testing-library/react'

import AvatarInfo from '.'

describe('AvatarInfo Test', () => {
  test('Load AvatarInfo', () => {
    const { container } = render(
      <AvatarInfo
        image="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcTQEZrATmgHOi5ls0YCCQBTkocia_atSw0X-Q&usqp=CAU"
        title="Dr Jonh Smith"
        text="Lorem ipsum dolor sit amet consectetur adipisicing elit. Adipisci quod doloribus culpa?"
      />
    )
    expect(container).toMatchSnapshot()
  })
})
