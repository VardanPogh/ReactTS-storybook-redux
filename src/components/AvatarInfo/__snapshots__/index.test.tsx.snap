// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarInfo Test Load AvatarInfo 1`] = `
<div>
  <div
    class="MuiAvatar-root MuiAvatar-circle makeStyles-large-1"
  >
    <img
      class="MuiAvatar-img"
      src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcTQEZrATmgHOi5ls0YCCQBTkocia_atSw0X-Q&usqp=CAU"
    />
  </div>
  <p
    class="makeStyles-p_name-2"
  >
    Dr Jonh Smith
  </p>
  <p
    class="makeStyles-p_des-3"
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Adipisci quod doloribus culpa?
  </p>
</div>
`;
